lib/airbrake-ruby/backtrace.rb

```diff
@@ -188,6 +193,21 @@ def match_frame(regexp, stackframe)
 
         Patterns::GENERIC.match(stackframe)
       end
+
+      def read_code_hunk(logger, frame)
+        code_hunk = Airbrake::CodeHunk.new(frame[:file], frame[:line]).to_h
+        return unless code_hunk
+
+        unless code_hunk.key?(0)
```

Comment:
I see 2 `unless` that check for `code_hunk`. I also don't understand what `code_hunk.key?(0)` does so it would be nice if you refactor this code or make it more obvious.

Reply:
`#to_h` returns a hash of `int => string`. A file can never have a zero line. I thought the existence of it would mean there's an error.

First `unless` is when a file is empty. Second `unless` is when there's an error that we need to log.

Does that make sense?

Reply:
It makes sense, but why not to throw an exception like any other Ruby code?

Reply:
It would be unfortunate to miss a notice just because we couldn't read a line of code. I think simply ignoring that error and moving on would be the best solution.

Reply:
I don't suggest you throw that exception to users. You can catch and log it as usually.

Reply:
👍